### PR TITLE
fix output of model

### PR DESCRIPTION
**Pull request by amal-ghamdi**
closes https://github.com/CUQI-DTU/CUQIpy-FEniCS/issues/35

The related CUQIpy-FEniCS issue needed to be fixed in the CUQIpy repo

### CODE REVIEW

File: cuqi/model/_model.py [hunk 1]
```diff
@@ -148,57 +148,82 @@ def domain_dim(self):
...
         ndarray or cuqi.array.CUQIarray
-            The input value represented as a function.
+            `x` represented as a function.
         """
-        if type(x) is CUQIarray and not isinstance(x.geometry, _DefaultGeometry1D):
-            return x.funvals
+        # Convert to function representation
+        # if x is CUQIarray and geometry are consistent, we obtain funvals
+        # directly
+        if isinstance(x, CUQIarray) and  x.geometry == geometry:
```
Comment (nabriis) [inline cuqi/model/_model.py:178]: Just to be sure. Can we compare the compute time when input is CUQIarray for Models now? Is the geometry check causing a lot of overhead?

**Review by nabriis — COMMENTED** [review]
Looks good to me. I raise one concern relating to the potential extra overhead introduced when input is CUQIarray to Model forward evaluation

**Review by jakobsj — APPROVED** [review]
Looks good, no additional comments, but good point raised by @nabr

**Comment by amal-ghamdi** [comment]
Thank you @nabriis and @jakobsj for your review. It is a good point about the overhead. I documented the overhead for a couple of examples in the attached file. It seems alright, but let me know what you think.


[overhead_geom_check.pdf](https://github.com/CUQI-DTU/CUQIpy/files/12494563/overhead_geom_check.pdf)

**Review by nabriis — APPROVED** [review]

**Pull request merged**